presto-main/src/main/java/com/facebook/presto/operator/scalar/JsonFunctions.java

```diff
@@ -88,6 +88,29 @@ public static JsonPath castCharToJsonPath(@LiteralParameter("x") Long charLength
         return new JsonPath(padSpaces(pattern, charLength.intValue()).toStringUtf8());
     }
 
+    @SqlNullable
+    @ScalarFunction("is_json_scalar")
+    @LiteralParameters("x")
+    @SqlType(StandardTypes.BOOLEAN)
+    public static Boolean varcharIsJsonScalar(@SqlType("varchar(x)") Slice json)
+    {
+        return isJsonScalar(jsonParse(json));
```

Comment:
Don't need to call `jsonParse` here.

Reply:
We have to call `jsonParse` because we need to make sure if this varchar a valid json before checking if it is a scalar.

Reply:
We can do valid json check in `isJsonScalar`, like the following code (assume we return null instead of throw `INVALID_FUNCTION_ARGUMENT`:

```
        try (JsonParser parser = createJsonParser(JSON_FACTORY, json)) {
            JsonToken token = parser.nextToken();
            if (token == null) {
                return null;
            }

            if (token == START_ARRAY) {
                parser.skipChildren();
                if (parser.nextToken() != null) {
                    // trailing token
                    return null;
                }
                return false;
            }
            else if (token == START_OBJECT) {
                parser.skipChildren();
                if (parser.nextToken() != null) {
                    // trailing token
                    return null;
                }
                return false;
            }
            else {
                if (parser.nextToken() != null) {
                    // trailing token
                    return null;
                }
                return true;
            }
        }
        catch (IOException e) {
            return null;
        }
```